theano/gof/fg.py

```diff
@@ -478,8 +481,14 @@ def find_path_to(output_var, input_var):
 
         for node in new_nodes:
             assert node not in self.apply_nodes
-            self.__setup_node__(node)
-            self.apply_nodes.add(node)
+            prevent_addition = False
+            for n in self._removed_nodes :
+                if node is n :
+                    prevent_addition = True
+                    
+            if not prevent_addition : 
```

Comment:
There is a checkpoint mechanism in the optimization. So if an optimization cause a problem, we revert to the state before and skip it. So yes, we can raise an error here. This is used for inplace optimization, so this is a used, tested and working system in Theano.